src/utils/request.ts

```diff
@@ -14,3 +28,82 @@ export function parseUrl(request: Request): URL | undefined {
 
   return url;
 }
+
+export function parseConditionalHeaders(headers: Headers): ConditionalHeaders {
+  const ifModifiedSince = headers.has('if-modified-since')
```

Comment:
```js
const ifModifiedSince = headers.has('if-modified-since') &&
  new Date(headers.get('if-modified-since');

if (ifModifiedSince instanceof Date) {
  ifModifiedSince.setSeconds(ifModifiedSince.getSeconds() + 1);
}
```